tls/s2n_resume.c

```diff
@@ -261,30 +299,330 @@ int s2n_connection_get_session(struct s2n_connection *conn, uint8_t *session, si
 
     uint32_t len = s2n_connection_get_session_length(conn);
 
+    if (!len) {
+        return len;
+    }
+
     S2N_ERROR_IF(len > max_length, S2N_ERR_SERIALIZED_SESSION_STATE_TOO_LONG);
 
-    struct s2n_blob serailized_data;
-    serailized_data.data = session;
-    serailized_data.size = len;
-    GUARD(s2n_blob_zero(&serailized_data));
+    struct s2n_blob serialized_data;
+    serialized_data.data = session;
+    serialized_data.size = len;
+    GUARD(s2n_blob_zero(&serialized_data));
 
     struct s2n_stuffer to;
-    GUARD(s2n_stuffer_init(&to, &serailized_data));
+    GUARD(s2n_stuffer_init(&to, &serialized_data));
     GUARD(s2n_client_serialize_resumption_state(conn, &to));
 
     return len;
 }
 
 ssize_t s2n_connection_get_session_length(struct s2n_connection *conn)
 {
-    /* Since we only support session ids for now, return length as: "format + session_id_len + session_id + session state",
-     * needs to be updated once session tickets support is added.
+    /* Session resumption using session id: "format (0) + session_id_len + session_id + session state"
+     * or
+     * Session resumption using session ticket "format (1) + ticket_lifetime_hint + session_ticket_len + session_ticket + session state"
      */
-    return 1 + 1 + conn->session_id_len + S2N_STATE_SIZE_IN_BYTES;
+    if (conn->config->use_tickets && conn->client_ticket.size > 0) {
+        return 1 + 4 + 2 + conn->client_ticket.size + S2N_STATE_SIZE_IN_BYTES;
```

Comment:
probably move these into constants + use sizeof?

Reply:
Have changed this to use macros.